src/predictor/gpu_predictor.cu

```diff
@@ -361,10 +364,14 @@ class GPUPredictor : public xgboost::Predictor {
       DeviceOffsets(batch.offset, batch.data.Size(), &device_offsets);
       batch.data.Reshard(GPUDistribution::Explicit(devices_, device_offsets));
 
-      // TODO(rongou): only copy the model once for all the batches.
+      if (batch_offset == 0) {
```

Comment:
can this be hoisted out of the for loop?

Reply:
Done.